src/swap/hw-app-swap/Swap.js

```diff
@@ -0,0 +1,211 @@
+// @flow
+import type Transport from "@ledgerhq/hw-transport";
+
+const START_NEW_TRANSACTION_COMMAND: number = 0x01;
+const SET_PARTNER_KEY_COMMAND: number = 0x02;
+const CHECK_PARTNER_COMMAND: number = 0x03;
+const PROCESS_TRANSACTION_RESPONSE: number = 0x04;
+const CHECK_TRANSACTION_SIGNATURE: number = 0x05;
+const CHECK_PAYOUT_ADDRESS: number = 0x06;
+const CHECK_REFUND_ADDRESS: number = 0x07;
+
+export default class Swap {
+  transport: Transport<*>;
+  allowedStatuses: Array<number> = [
```

Comment:
I'm not exactly understanding why we need to **allow** these statuses. if there are error cases, we should let them throw and not ignore them, is there something i'm missing? this should be commented/documented.